collector.go

```diff
@@ -220,34 +221,66 @@ func countJobs(run *github.WorkflowRun, jobs []*github.WorkflowJob) {
 			}
 			stepRunTime := step.CompletedAt.Time.Sub(step.StartedAt.Time)
 			stepRunTimeVec.WithLabelValues(
-				repo, ref, workflowName, job.GetName(), step.GetName(),
+				repo, ref, eventType, workflowName, job.GetName(), step.GetName(),
 			).Add(stepRunTime.Seconds())
 			jobRunTime += stepRunTime
 		}
 		jobRunCountVec.WithLabelValues(
-			repo, ref, workflowName, job.GetName(), job.GetConclusion(),
+			repo, ref, eventType, workflowName, job.GetName(), job.GetConclusion(),
 		).Add(1)
 		if job.GetConclusion() != "success" {
 			continue
 		}
 		jobRunTimeVec.WithLabelValues(
-			repo, ref, workflowName, job.GetName(),
+			repo, ref, eventType, workflowName, job.GetName(),
 		).Add(jobRunTime.Seconds())
 
 		workflowRunTime += jobRunTime
 	}
 
-	workflowRunCountVec.WithLabelValues(repo, ref, workflowName, run.GetConclusion()).Add(1)
+	workflowRunCountVec.WithLabelValues(repo, ref, eventType, workflowName, run.GetConclusion()).Add(1)
 
 	if run.GetConclusion() != "success" {
 		return
 	}
 
-	workflowRunTimeVec.WithLabelValues(repo, ref, workflowName).Add(workflowRunTime.Seconds())
+	workflowRunTimeVec.WithLabelValues(repo, ref, eventType, workflowName).Add(workflowRunTime.Seconds())
 }
 
 func makeRef(run *github.WorkflowRun) string {
-	return "todo"
+	if strings.HasPrefix(run.GetEvent(), "pull_request") {
+		// Attempt to tie the workflow to a PR
+		headSha := run.GetHeadSHA()
+		headBranch := run.GetHeadBranch()
+
+		if headSha == "" && headBranch == "" {
+			return "error-missing-head-ref"
+		}
+
+		for _, pr := range run.PullRequests {
+			prBaseBranch := pr.GetBase()
+			if prBaseBranch == nil {
+				continue
+			}
+
+			if prBaseBranch.GetSHA() == headSha {
+				return headSha
+			}
+
+			if prBaseBranch.GetRef() == headBranch {
+				return headBranch
+			}
+		}
+
+		return "error-no-matching-pr"
+	}
+
+	headBranch := run.GetHeadBranch()
```

Comment:
I'm wondering if this is what we want for merge_group events - I assume for merge_group that the pull requests would be valid in the run, so perhaps the above logic for pull requests should also apply to merge group events. 

I'm still waiting to find a merge_group PR with `pull_requests` non-empty - but an alternative is to parse the branch out of the head branch which is of the form "gh-readonly-queue/master/pr-39050-deeadc4b4cee7de73be618a1b612955ea6006d0b"